tests/perf-test/OptionsListUtils.perf-test.ts

```diff
@@ -0,0 +1,78 @@
+import {rand} from '@ngneat/falso';
+import type * as NativeNavigation from '@react-navigation/native';
+import {measureFunction} from 'reassure';
+import * as OptionsListUtils from '@libs/OptionsListUtils';
+import CONST from '@src/CONST';
+import ONYXKEYS from '@src/ONYXKEYS';
+import type {PersonalDetails} from '@src/types/onyx';
+import type Report from '@src/types/onyx/Report';
+import createCollection from '../utils/collections/createCollection';
+import createPersonalDetails from '../utils/collections/personalDetails';
+import {getRandomDate} from '../utils/collections/reportActions';
+import createRandomReport from '../utils/collections/reports';
+import waitForBatchedUpdates from '../utils/waitForBatchedUpdates';
+
+const REPORTS_COUNT = 5000;
+const PERSONAL_DETAILS_LIST_COUNT = 1000;
+const SEARCH_VALUE = 'TestingValue';
+
+const reports = createCollection<Report>(
+    (item) => `${ONYXKEYS.COLLECTION.REPORT}${item.reportID}`,
+    (index) => ({
+        ...createRandomReport(index),
+        type: rand(Object.values(CONST.REPORT.TYPE)),
+        lastVisibleActionCreated: getRandomDate(),
+    }),
+    REPORTS_COUNT,
+);
+
+const personalDetails = createCollection<PersonalDetails>(
+    (item) => item.accountID,
+    (index) => createPersonalDetails(index),
+    PERSONAL_DETAILS_LIST_COUNT,
+);
+
+const mockedBetas = Object.values(CONST.BETAS);
+
+jest.mock('@react-navigation/native', () => {
+    const actualNav = jest.requireActual('@react-navigation/native');
+    return {
+        ...actualNav,
+        createNavigationContainerRef: () => ({
+            getState: () => jest.fn(),
+        }),
+    } as typeof NativeNavigation;
+});
+
+/* GetOption is the private function and is never called directly, we are testing the functions which call getOption with different params */
+describe('OptionsListUtils', () => {
+    /* Testing getSearchOptions */
+    test('[OptionsListUtils] getSearchOptions with search value', async () => {
+        await waitForBatchedUpdates();
+        await measureFunction(() => OptionsListUtils.getSearchOptions(reports, personalDetails, SEARCH_VALUE, mockedBetas));
```

Comment:
it's going to most of the IFs which are triggered but that functions:
```
getSearchOptions
getShareLogOptions
getFilteredOptions
getShareDestinationOptions
getMemberInviteOptions
```

I didn't add tests to run `getOption` individually, because it is not called directly in the codebase. So it might not be relevant. Do you think we add those?

Reply:
Makes sense!